hs_file_types/models/reftimeseries.py

```diff
@@ -799,6 +799,16 @@ def is_single_file_aggregation(self):
         """This aggregation supports only one file"""
         return True
 
+    @property
+    def redirect_url(self):
```

Comment:
I am not sure how this property method is being used. Is this property relevant only to the aggregation that can contain only one file? I was wondering if we can move this to the base aggregation class ```AbstractLogicalFile```.

Reply:
Pretty sure redirect_url is used for referenced file content.  This should probably be resolved in the download code, not just added to all logical files.

Reply:
I propose updating this block to check fore the redirect_url property before trying to access it.
https://github.com/hydroshare/hydroshare/blob/develop/django_irods/views.py#L142

Reply:
From what I can see, the redirect_url property is only used if the aggregation is a single file aggregation. In this block it first checks if the file is a single file aggregation before referencing the redirect_url.

https://github.com/hydroshare/hydroshare/blob/897831aceb70bef6697f5f07bbcd28a16c207365/django_irods/views.py#L137-L144

The property already existed in GenericLogicalFile, but not in RefTimeseriesLogicalFile, which is what caused the error only for those aggregations. Are there other single file aggregations we should add the redirect_url property to?

https://github.com/hydroshare/hydroshare/blob/8f2088b5c15c1332de43a1ad7e1d7d74b4cbad75/hs_file_types/models/generic.py#L211-L219

I can update that block to check if the redirect_url property exists so it doesn't raise an attribute error, but I think the property needs to exist to be able to download the file.

Reply:
No, that property isn't needed to download the file.  That property, if it exists will redirect to the url.  It's for referenced file content.

Reply:
It can be hard to read as the code isn't very intuitive.

Reply:
Alright, commit 3c39e96909b4f0682793aa9540a1f34d534d2dcf checks if the redirect_url exists instead of adding the property to RefTimeseriesLogicalFile.